src/CmdExecuteService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Islandora\Crayfish\Commons;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Runs a command streaming data in on stdin and out on stdout.
+ *
+ * @package Islandora\Crayfish\Commons
+ */
+class CmdExecuteService
+{
+
+    /**
+     * @var null|\Psr\Log\LoggerInterface
+     */
+    protected $log;
+
+    /**
+     * Executor constructor.
+     * @param LoggerInterface $log
+     */
+    public function __construct(LoggerInterface $log = null)
+    {
+        $this->log = $log;
+    }
+
+    /**
+     * Runs the command
+     *
+     * @param $cmd
+     * @param $data
+     *
+     * @throws \RuntimeException
+     *
+     * @return \Closure
+     *   Closure that streams the output of the command.
+     */
+    public function execute($cmd, $data)
```

Comment:
I pushed the arguments up to the parent class, so this one could be used more generically. You just pass a full command to run here. 

You can see how this is used in Hypercube here:
https://github.com/Islandora-CLAW/Crayfish/pull/19/commits/fadd616b35cf577f494e1165033f58609a0bab5a